cider/exec/nextgen/jitlib/base/JITModule.h

```diff
@@ -86,7 +86,7 @@ class JITFunctionBuilder {
 
     JITFunctionPointer function = module_->createJITFunction(JITFunctionDescriptor{
         .function_name = name_, .ret_type = return_, .params_type = parameter_list_});
-    builder_(function.get());
+    builder_(function);
```

Comment:
Why changed ```builder_```'s signature to shared_ptr? In ```JITFunctionBuilder```, ```builder_``` will be called here only, and always at the same block with ```function```, so there is no lifetime problem.

Reply:
As builder only save the raw pointer of function, so the function will be released after the build, need to save the function somewhere

Reply:
I see. Maybe change function signature to ```void(const jitlib::JITFunctionPointer&)``` is better.